### PR TITLE
Use broccoli-funnel instead of broccoli-static-compiler

### PR DESCRIPTION
**Pull request by joliss**
broccoli-funnel has the `allowEmpty` option that we needed CustomStaticCompiler for.

I didn't get any failing tests, so I'm not 100% sure that the code in `lib/models/addon.js` works properly, but it definitely *looks* correct. :)

### CODE REVIEW

**Comment by stefanpenner** [comment]
awesome, by sure though that: `npm run test-all` was run.

**Comment by stefanpenner** [comment]
tests failures look race related. I suspect we recently regressed on something related... in all likely-hood one of or deps has a race again...

**Comment by stefanpenner** [comment]
this appears to be a duplicate of: https://github.com/ember-cli/ember-cli/pull/3621

**Comment by joliss** [comment]
Ah, yes, now I'm getting the tests to cover my code. So things are OK then. Travis is green too.

Re dup, I caught a few extra lines I think, so you should merge my PR :)

**Comment by stefanpenner** [comment]
Ya. Done :)

**Comment by stefanpenner** [comment]
Gotta figure out those races. These brittle tests have been getting worse this week

**Comment by stefanpenner** [comment]
ah, I forgot add-ons depend on having access to `this.pickFiles`, we will likely need to add backwards compat

File: lib/models/addon.js [hunk 1]
```diff
@@ -364,7 +363,7 @@ Addon.prototype.treeForPublic = function(tree) {
     return tree;
   }
 
-  return this.pickFiles(tree, {
```
Comment (stefanpenner) [inline lib/models/addon.js]: specifically this case.

**Comment by stefanpenner** [comment]
https://github.com/ember-cli/ember-cli/pull/3675

**Pull request merged**